packages/core/src/StateMachine.ts

```diff
@@ -442,9 +452,53 @@ export class StateMachine<
       State<TContext, TEvent, TResolvedTypesMeta>
     >
   ): State<TContext, TEvent, TResolvedTypesMeta> {
-    const restoredState = isStateConfig(state)
-      ? this.resolveState(state as any)
-      : this.resolveState(State.from(state as any, this.context, this));
+    let restoredState: State<TContext, TEvent, TResolvedTypesMeta>;
```

Comment:
q: what should happen if we rehydrate a machine with a state that is missing some child while the machine configuration expects that child to exist (like if we'd rehydrate in `b` that has `invoke: 'fooActor'` without a persisted state for that `fooActor`)?

Reply:
I think by default, we would just start (invoke) the `'fooActor'`.

Reply:
we need to add a test case for this

Reply:
Added a skipped test case; may be best handled after this PR